src/engraving/libmscore/score.cpp

```diff
@@ -5393,33 +5393,30 @@ void Score::changeSelectedNotesVoice(voice_idx_t voice)
     setLayoutAll();
 }
 
-std::set<ID> Score::partIdsFromRange(const track_idx_t trackFrom, const track_idx_t trackTo) const
+std::set<staff_idx_t> Score::staffIdsFromRange(const track_idx_t trackFrom, const track_idx_t trackTo, StaffAccepted staffAccepted) const
 {
-    std::set<ID> result;
+    std::set<staff_idx_t> result;
 
-    for (const Part* part : m_score->parts()) {
-        if (trackTo < part->startTrack() || trackFrom >= part->endTrack()) {
+    for (staff_idx_t idx = 0; idx < _staves.size(); ++idx) {
+        const Staff* staff = _staves.at(idx);
+        if (!staff) {
             continue;
         }
 
-        result.insert(part->id());
-    }
-
-    return result;
-}
-
-std::set<staff_idx_t> Score::staffIdsFromRange(const track_idx_t trackFrom, const track_idx_t trackTo) const
-{
-    std::set<staff_idx_t> result;
+        track_idx_t startTrackIdx = staff2track(idx, 0);
+        track_idx_t endTrackIdx = startTrackIdx + VOICES;
 
-    for (const Part* part : m_score->parts()) {
-        if (trackTo < part->startTrack() || trackFrom >= part->endTrack()) {
+        if (startTrackIdx > trackTo || endTrackIdx <= trackFrom) {
```

Comment:
I wonder if you can eliminate this check by altering the for-loop to something like this:
```c++
staff_idx_t staffIdxFrom = trackFrom / VOICES;
staff_idx_t staffIdxTo = std::min(trackTo / VOICES + VOICES, _staves.size());

for (staff_idx_t idx = staffIdxFrom; idx < staffIdxTo; ++idx) {
    …
```

Reply:
Thanks, this looks better